modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ExecutionServiceImpl.java

```diff
@@ -578,19 +586,12 @@ private CompletableFuture<Void> executeFragment(IgniteRel treeRoot, ResolvedDepe
             if (!(node instanceof Outbox)) {
                 Function<RowT, RowT> internalTypeConverter = TypeUtils.resultTypeConverter(ectx, treeRoot.getRowType());
 
-                AsyncRootNode<RowT, List<Object>> rootNode = new AsyncRootNode<>(node, inRow -> {
-                    inRow = internalTypeConverter.apply(inRow);
-
-                    int rowSize = ectx.rowHandler().columnCount(inRow);
-
-                    List<Object> res = new ArrayList<>(rowSize);
-
-                    for (int i = 0; i < rowSize; i++) {
-                        res.add(ectx.rowHandler().get(i, inRow));
-                    }
-
-                    return res;
-                });
+                AsyncRootNode<RowT, InternalSqlRow> rootNode = new AsyncRootNode<>(
+                        node,
+                        inRow -> {
+                            inRow = internalTypeConverter.apply(inRow);
```

Comment:
it still required for Date/Time types at least.  Otherwise we have uncovered from internal representation values

Reply:
still, this doesn't feel right. Rows created by RowFactory are supposed to work with internal format only. Everything is forking now is because we still have ObjectsArrayRowWrapper which can store arbitrary values regardless the types.

proper way is to have additional conversion in InternalSqlRowImpl. See usages of `org.apache.ignite.internal.sql.engine.util.TypeUtils#fromInternal` for example